lib/api/controllers/remoteActions/managePlugins.js

```diff
@@ -0,0 +1,591 @@
+                }
+                else if (needInstall(pluginsInfos[name], name)) {
+                  pluginsList.push(clcNotice(name + ' (not installed)'));
+                }
+                else {
+                  pluginsList.push(clcNotice(name + ' (disabled)'));
+                }
+              });
+              deferred.resolve(pluginsList);
+            });
+        }
+        if (options.install) {
+          if (plugin) {
+            console.log('███ kuzzle-plugins: Installing plugin ' + plugin + '...');
+          }
+          else {
+            console.log('███ kuzzle-plugins: Starting plugins installation...');
+          }
+
+          return installPlugins(plugin, options, dbService, kuzzleConfiguration);
+        }
+
+        if (options.get) {
+          return getPluginConfiguration(plugin, dbService, kuzzleConfiguration);
+        }
+
+        if (options.set) {
+          return setPluginConfiguration(plugin, options.set, dbService, kuzzleConfiguration);
+        }
+
+        if (options.importConfig) {
+          return importPluginConfiguration(dbService, plugin, options.importConfig, kuzzleConfiguration);
+        }
+
+        if (options.unset) {
+          return unsetPluginConfiguration(plugin, options.unset, dbService, kuzzleConfiguration);
+        }
+
+        if (options.replace) {
+          return replacePluginConfiguration(plugin, options.replace, dbService, kuzzleConfiguration);
+        }
+
+        if (options.remove) {
+          return removePlugin(plugin, dbService, kuzzleConfiguration);
+        }
+
+        if (options.activate) {
+          return dbService
+            .update(kuzzleConfiguration.pluginsManager.dataCollection, plugin, {activated: true})
+            .then(() => getPluginConfiguration(plugin, dbService, kuzzleConfiguration));
+        }
+
+        if (options.deactivate) {
+          return dbService
+            .update(kuzzleConfiguration.pluginsManager.dataCollection, plugin, {activated: false})
+            .then(() => getPluginConfiguration(plugin, dbService, kuzzleConfiguration));
+        }
+      })
+      .then((res) => {
+        release();
+
+        if (options.install) {
+          if (plugin) {
+            console.log(clcOk('███ kuzzle-plugins: Plugin ' + plugin + ' installed'));
+          }
+          else {
+            console.log(clcOk('███ kuzzle-plugins: Plugins installed'));
+          }
+        }
+        deferred.resolve(res);
+      })
+      .catch(error => {
+        release();
+
+        /*
+         If elasticsearch returns with a "Service temporary unavailable" error,
+         we retry until it's ready
+         */
+        if (error.status === 503) {
+          console.log('The database seems to not be ready yet. Retrying in 5s...');
+          return setTimeout(() => pluginsManager(plugin, options), 5000);
+        }
+        deferred.reject(error);
+      });
+  });
+  return deferred.promise;
```

Comment:
OK, forgot my remarks, I did not notice that we were inside a callback function :/